### PR TITLE
feat(codegen): add Integer#allbits?

### PR DESCRIPTION
**Pull request by Halvanhelv**
This adds `Integer#allbits?` — returns true when all bits in the mask are set in the receiver.

The codegen emits a simple bitwise AND comparison: `(self & mask) == mask`. No runtime function needed.

Tests covering edge cases (zero mask, subset, no match, negative receiver, large values) are in `test/integer_allbits.rb`.

### CODE REVIEW

File: spinel_codegen.rb [hunk 1]
```diff
@@ -18425,6 +18428,9 @@ def compile_int_method_expr(nid, mname, rc)
     if mname == "odd?"
       return "((" + rc + ") % 2 != 0)"
     end
+    if mname == "allbits?"
+      return "(((" + rc + ") & (" + compile_arg0(nid) + ")) == (" + compile_arg0(nid) + "))"
+    end
```
[the remark this answers is not in the document]
Comment (Halvanhelv) [reply spinel_codegen.rb:18434]: make sense, updated
